source/initial_temperature/patch_on_S40RTS.cc

```diff
@@ -0,0 +1,197 @@
+/*
+  Copyright (C) 2011 - 2017 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <aspect/global.h>
+#include <aspect/adiabatic_conditions/interface.h>
+#include <aspect/initial_temperature/patch_on_S40RTS.h>
+#include <aspect/utilities.h>
+
+namespace aspect
+{
+  namespace InitialTemperature
+  {
+    template <int dim>
+    PatchOnS40RTS<dim>::PatchOnS40RTS ()
+    {}
+
+
+    template <int dim>
+    void
+    PatchOnS40RTS<dim>::initialize ()
+    {
+      Utilities::AsciiDataInitial<dim>::initialize(1);
+      if (s40rts.vs_to_density_method == s40rts.file)
+        {
+          s40rts.profile.initialize(this->get_mpi_communicator());
+          s40rts.vs_to_density_index = s40rts.profile.get_column_index_from_name("vs_to_density");
+        }
+    }
+
+
+    template <int dim>
+    double
+    PatchOnS40RTS<dim>::
+    ascii_grid_vs (const Point<dim> &position) const
+    {
+      const double vs_perturbation = Utilities::AsciiDataInitial<dim>::get_data_component(position,0);
+      return vs_perturbation;
+    }
+
+
+    template <int dim>
+    double
+    PatchOnS40RTS<dim>::
+    initial_temperature (const Point<dim> &position) const
+    {
+      const double depth = this->get_geometry_model().depth(position);
+
+      double vs_perturbation;
+      if (depth <= max_grid_depth - smoothing_length_scale)
+        {
+          vs_perturbation = ascii_grid_vs(position);
+        }
+      //add smoothing between the two models
+      else if (depth > max_grid_depth - smoothing_length_scale && depth < max_grid_depth)
+        {
+          const double scale_factor = (depth-(max_grid_depth-smoothing_length_scale))/smoothing_length_scale;
+          vs_perturbation = s40rts.get_Vs(position)*(scale_factor) + ascii_grid_vs(position)*(1-scale_factor);
```

Comment:
Thanks for putting in the smoothing!

Although the integer 1 is promoted to a double here automatically, I usually write these numbers as doubles (1.0) for clarity.